code/modules/spells/targeted/recall.dm

```diff
@@ -0,0 +1,176 @@
+/spell/targeted/bound_object
+	name = "Bound Object"
+	desc = "This spell allows a wizard to bind an object to themselves, then teleport it to them at will. Middle click the spell icon or use the 'Unbind' spell to select a new object."
+	abbreviation = "BO"
+
+	school = "abjuration"
+	charge_max = 100
+	spell_flags = SELECTABLE | WAIT_FOR_CLICK
+	hud_state = "wiz_bound"
+	level_max = list(Sp_TOTAL = 3, Sp_SPEED = 2, Sp_POWER = 1)
+
+	var/has_object = 0
+	var/obj/bound
+	var/icon/bound_icon
+
+	var/allow_anchored = 0
+
+	var/static/list/prohibited = list( //Items that are prohibited because, frankly, it would cause more unfun for everyone else than fun for the user if they could be retrieved.
+		/obj/machinery/power/apc,								//APCs
+		/obj/machinery/atmospherics,							//pipes, vents, pumps, the cryo tubes, the gas miners, etc.
+		/obj/machinery/alarm,									//air alarms
+		/obj/machinery/firealarm,								//fire alarms
+		/obj/machinery/status_display,							//status displays
+		/obj/machinery/newscaster,								//newscasters
+		/obj/item/device/radio/intercom,						//intercoms
+		/obj/structure/extinguisher_cabinet,					//fire extinguisher cabinets
+		/obj/machinery/computer/security/telescreen,			//TV screens
+		/obj/machinery/camera,									//AI cameras
+		/obj/machinery/requests_console,						//requests consoles
+		/obj/machinery/door_control,							//door control buttons
+		/obj/structure/closet/fireaxecabinet,					//fire axe cabinets
+		/obj/machinery/light_switch,							//light switches                  //list taken from subspacetunneler.dm
+		/obj/structure/sign,									//area signs
+		/obj/structure/closet/walllocker,						//defib lockers, wall-mounted O2 lockers, etc.
+		/obj/machinery/recharger/defibcharger/wallcharger,		//wall-mounted defib chargers
+		/obj/structure/noticeboard,								//notice boards
+		/obj/machinery/space_heater/campfire/stove/fireplace,	//fireplaces
+		/obj/structure/painting,								//paintings
+		/obj/item/weapon/storage/secure/safe,					//wall-mounted safes
+		/obj/machinery/door_timer,								//brig cell timers
+		/obj/structure/closet/secure_closet/brig,				//brig cell closets
+		/obj/machinery/disposal,								//disposal bins
+		/obj/machinery/light,									//light bulbs and tubes
+		/obj/machinery/sleep_console,							//sleeper consoles
+		/obj/machinery/sleeper,									//sleepers
+		/obj/machinery/body_scanconsole,						//body scanner consoles
+		/obj/machinery/bodyscanner,								//body scanners
+		/obj/machinery/media/receiver/boombox/wallmount,		//sound systems
+		/obj/machinery/keycard_auth,							//keycard authentication devices
```

Comment:
Shards will irradiate the shit out of the wizard if they grab it, and singularities are anchored so they'll need to spend two spell points on what is basically a suicide spell